Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM node:18-alpine as build
+
+RUN mkdir -p /opt/alertra-promtheus-exporter
+WORKDIR /src
+
+COPY package*.json tsconfig.json app.ts ./
+COPY lib lib/
+RUN tree ./
+RUN npm install --unsafe-perm
+RUN npm exec tsc
+
+FROM node:18-alpine
+WORKDIR /opt/alertra-prometheus-exporter
+COPY --from=build /src/package*.json /src/dist ./
+RUN tree ./
+RUN npm ci --omit=dev && npm cache clean --force && find -name "*.ts" -delete
+RUN tree ./
+ENV PORT=13964
+EXPOSE 13964
+ENTRYPOINT ["node", "app.js"]
```

Comment:
Im not sure i understand this Dockerfile. What are we optimizing for here?

I ask because the layout here (by eyeball) looks like it would have slow REPL time for changes. For example, if we have `build` stage that `dev` and `prod`/`run` are further built from, i wouldnt expect us to need `package*` in either.

Reply:
> What are we optimizing for here?

Size I suppose. I was annoyed that I had < 10kb of code that resulted in *many* MB of docker layers, so I googled around a bit to see what others had done.

No, it's not critical as I don't expect to be pushing this image anywhere, it did bring the size down to ~7MB or so, but yes, it doesn't really matter.

You're right, it does slow down the time needed for building (to like 15-20s). It was an interesting experiment, trimming things I knew we didn't need at runtime (typescript, type packages, npm cache, ...).

Reply:
The important thing, especially when developing these new container specs, is to keep the dev REPL quick. If we dont change the `package*`, lets include them in a layer early, with `npm install` or `npm ci`. Having to rebuild package dependencies when the package spec hasnt changed is probably the slowest and most frustrating part of iterating on a container image. If only the code has changed, lets make sure to not cache bust earlier layers.